pilot/pkg/model/xds_cache.go

```diff
@@ -117,7 +112,6 @@ func NewXdsCache() XdsCache {
 	cache := &lruCache{
 		enableAssertions: features.EnableUnsafeAssertions,
 		configIndex:      map[ConfigHash]sets.String{},
-		typesIndex:       map[kind.Kind]sets.String{},
```

Comment:
What if instead of a map of `Kind -> key` -- which is huge, we have a map of kind -> KeyType?

So `PeerAuthentication: EndpointBuilder`.

That way the cost is basically zero to maintain, but we get all the same benefits

Reply:
I am not sure then how we can clear the cache when pa changes?

Reply:
I think we would need to have a multi level map I guess if keytype->map

instead of all different key types shared in one map

Maybe not worth it

Reply:
This is resolved by https://github.com/istio/istio/pull/43428#discussion_r1111402466

Reply:
Sharding to multiple maps may also help lock contention if we think that is a bottleneck